src/main/java/uk/gov/hmcts/reform/sscs/hearing/mapping/PartyFlagsMapping.java

```diff
@@ -0,0 +1,133 @@
+package uk.gov.hmcts.reform.sscs.hearing.mapping;
+
+import uk.gov.hmcts.reform.sscs.ccd.domain.Appeal;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingOptions;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.YesNo;
+import uk.gov.hmcts.reform.sscs.model.servicehearingvalues.PartyFlags;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class PartyFlagsMapping {
+
+    public List<PartyFlags> getPartyFlags(SscsCaseData caseData) {
+        List<PartyFlags> partyFlagsList = new ArrayList<>();
+        partyFlagsList.add(mapSignLanguageType(caseData));
+        partyFlagsList.add(disabledAccess(caseData));
+        partyFlagsList.add(hearingLoop(caseData));
+        partyFlagsList.add(confidentialCase(caseData));
+        partyFlagsList.add(dwpUcb(caseData));
+        partyFlagsList.add(dwpPhme(caseData));
+        partyFlagsList.add(urgentCase(caseData));
+        partyFlagsList.add(adjournCaseInterpreterLanguage(caseData));
+        return partyFlagsList;
+    }
+
+    private PartyFlags mapSignLanguageType(SscsCaseData caseData) {
+        var signLanguageType = Optional
+               .ofNullable(caseData.getAppeal())
+               .map(Appeal::getHearingOptions)
+               .map(HearingOptions::getArrangements);
```

Comment:
HearingOptions actually has convenience methods for these (wantsSignLanguageInterpreter, wantsHearingLoop, wantsAccessibleHearingRoom) it would be simpler to use them

Reply:
In line 32 it looks like it should be ;
 .map(HearingOptions::getSignLanguageType);

Reply:
I can change it let me know.